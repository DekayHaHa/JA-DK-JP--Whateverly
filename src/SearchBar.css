.search-form {
  position: fixed;
  bottom: 0;
  width: 20%;
}

.search {
  width: 100%;
  height: 98px;
  /*background: none;*/
  border: none;
  color: white;
  font-size: 30px;
}

.search:focus {
  outline: none;
}

.search::placeholder {
  color: #234987;
  transition: all .2s ease;
}

.search:focus::placeholder {
  color: hsla(0, 0%, 0%, 0)
}

.clear-search {
  width: 20px;
  height: 20px;
  position: absolute;
  right: 5px;
  top: 40px;
}

.left-line,
.right-line {
  position: absolute;
  background: #506790;
  width: 5px;
  height: 20px;
}

.left-line { transform: rotate(45deg); }
.right-line { transform: rotate(-45deg); }