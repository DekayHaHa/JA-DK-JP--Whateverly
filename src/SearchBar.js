import React, {Component} from 'react';
import './SearchBar.css';

export default class SearchBar extends Component {
  constructor(props) {
    super(props)
    this.state = { }
  }
  updateText = event => {
    let searchText = event.target.value.toLowerCase()
    this.props.updateState({titleFilter: searchText})
  }

  render() {
    return(
      <div className="search-bar search-section">
        <form className="search-form">
          <input maxLength="25" placeholder="Search a game title!" className="search" onChange={this.updateText}/>
          <div className="clear-search">
            <div className="left-line"></div>
            <div className="right-line"></div>
          </div>
        </form>
      </div>
    )
  }
}