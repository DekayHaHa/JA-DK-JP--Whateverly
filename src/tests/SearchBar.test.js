import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from '../SearchBar';
import { shallow } from 'enzyme';

describe('SearchBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SearchBar />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
})