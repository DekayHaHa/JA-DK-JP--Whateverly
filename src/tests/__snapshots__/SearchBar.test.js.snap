// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshop 1`] = `
<div
  className="search-bar search-section"
>
  <form
    className="search-form"
  >
    <input
      className="search"
      maxLength="25"
      onChange={[Function]}
      placeholder="Search a game title!"
    />
    <div
      className="clear-search"
    >
      <div
        className="left-line"
      />
      <div
        className="right-line"
      />
    </div>
  </form>
</div>
`;
